autopwn.py

```diff
@@ -45,47 +35,76 @@ def StartScanning(
         Targets = [targetarg]
 
     ScanPorts, ScanVulns, DownloadExploits = UserConfirmation()
-    ScanWeb = WebScan()
+    PortArray = []
+
+    all_vulnerabilities = []
 
     for host in Targets:
         if ScanPorts:
             PortScanResults = PortScan(
-                host, log, args.speed, args.host_timeout, scanmode, args.nmap_flags
+                host, log, args.speed, args.host_timeout, scanmode, args.nmap_flags, shodan_api_key
             )
-            PortArray = AnalyseScanResults(PortScanResults, log, console, host)
-            if ScanVulns and len(PortArray) > 0:
-                VulnsArray = SearchSploits(PortArray, log, console, console2, apiKey)
-                if DownloadExploits and len(VulnsArray) > 0:
-                    GetExploitsFromArray(VulnsArray, log, console, console2, host)
-
-        if ScanWeb:
-            webvuln(host, log, console)
+            PortArray = AnalyseScanResults(PortScanResults, log, console, host, shodan_results=None)
+        if ScanVulns and PortArray and len(PortArray) > 0:
+            VulnsArray = []
+            keywords = generate_keywords(PortArray)
+            sploits = SearchSploits(keywords, log, console, args, apiKey)
+            for sploit in sploits:
+                all_vulnerabilities.append(sploit)
+            if shodan_api_key:
+                ShodanVulns, ShodanPorts = GetShodanVulns(host, shodan_api_key, log)
```

Comment:
The program shouldn't use Shodan and ZoomEye if the user is scanning local IP address/range.